packages/@ember/-internals/metal/lib/computed.ts

```diff
@@ -620,7 +626,117 @@ export default function computed(...args: (string | ComputedPropertyConfig)[]):
 
   return cp;
 }
+
+// https://tc39.github.io/proposal-decorators/#sec-elementdescriptor-specification-type
+interface ElementDescriptor {
+  descriptor: PropertyDescriptor;
+  initializer?: () => any; // unknown
+  key: string;
+  kind: 'method' | 'field' | 'initializer';
+  placement: 'own' | 'prototype' | 'static';
+}
+
+function requiresGetter(desc: ElementDescriptor) {
+  return ['method'].includes(desc.kind);
+}
+
+function computedDecoratorInner(dependents: any) {
+  return (desc: ElementDescriptor, params = []) => {
+    // All computeds are methods
+    // desc.kind = 'method';
+    desc.placement = 'prototype';
+
+    desc.finisher = function initializeComputedProperty(target) {
+      let { prototype } = target;
+      let { key } = desc;
+      let { get, set } = desc.descriptor;
+
+      if (requiresGetter(desc)) {
+        assert(`${key} is not a native accessor`, get);
+      }
+
+      let computedArgs = [...(dependents || []), {
+        get,
+        set
+      }];
+      let computedDesc = computed_original(...computedArgs);
+
+      defineProperty(prototype, key, computedDesc);
+
+      return target;
+    };
+  };
+}
+
+// computed(function() { ... })
+// computed('dependentKey', { get: function() { ... }})
+// computed('dependentKey', {})
+// computed('dependentKey', function() { ...})
+function isLegacyComputed(...args) {
```

Comment:
I don't believe this implementation will work fundamentally. Consider the following test case:

```js
let firstNameAlias = computed('firstName', function() {
  return this.firstName;
});

let Class1 = EmberObject.extend({
  otherFirstName: firstNameAlias
});

class Class2 {
  @firstNameAlias otherFirstName;
}
```

As specified in the RFC, this must work (and it must work for computed macros in general to work, since this is fundamental to them).

Reply:
I'll work on this case next, then :)